sonic_platform_base/sonic_sfp/ext_media_api.py

```diff
@@ -0,0 +1,200 @@
+########################################################################
+# DellEMC
+#
+# Module contains the initialization and main interface to consuming apps
+#
+########################################################################
+
+import importlib
+import inspect
+
+from ext_media_utils import DEFAULT_NO_DATA_VALUE
+from ext_media_common import form_factor_handler_to_ff_info, populate_eeprom_cache, get_form_factor_info
+from ext_media_handler_base import media_static_info
+from ext_media_cmis_init import INIT_TYPE_AUTO, DEFAULT_APPLICATION, cmis_init
+from ext_media_cmis_diag import cmis_diag
+
+"""
+Finds the appropriate driver module, executes calls
+Returns a dictionary of attributes from media, with default of DEFAULT_NO_DATA_VALUE is not able to get data.
+"""
+def get_static_info(sfp_obj, platform_obj=None):
+    eeprom_path = sfp_obj.get_eeprom_sysfs_path()
+    # First read 256 bytes for caching
+    eeprom_bytes = populate_eeprom_cache(eeprom_path, offset=0, length=256)
+    # Get an instance of the form factor handler, or return None (or throw) if error
+    def get_handler_instance():
+        if eeprom_bytes is None or len(eeprom_bytes) < 256:
+            return None
+
+        # Get the corresponding form-factor 
+        form_factor_name, form_factor_module  = get_form_factor_info(eeprom_bytes)
+        if None is form_factor_name:
+            return None
+        if None is form_factor_module:
+            raise NotImplementedError("Unable to find implementation for form-factor: " + str(form_factor_name))
+
+        form_factor_module_name = str(form_factor_module.__name__)
+        # Based on the form-factor, select the obj
+
+        # Get the classes
+        handler_class_list = inspect.getmembers(form_factor_module, lambda cl: inspect.isclass(cl))
+
+        handler_class = None
+
+        # Pick class which matches filename, less ext_media prefix
+        for cl_name, cl in handler_class_list:
+            if form_factor_module_name.split('ext_media_handler_')[1] ==  cl_name:
+                handler_class = cl
+                break
+        if handler_class is None:
```

Comment:
pls log an error message to indicate that the class with  required name is not defined in the media handler file for this xcvr type.